Tests/FluentAssertions.Specs/Specialized/TaskCompletionSourceAssertionSpecs.cs

```diff
@@ -9,8 +8,103 @@ namespace FluentAssertions.Specs.Specialized;
 
 public class TaskCompletionSourceAssertionSpecs
 {
+#if NET6_0_OR_GREATER
+
     [Fact]
     public async Task When_TCS_completes_in_time_it_should_succeed()
+    {
+        // Arrange
+        var subject = new TaskCompletionSource();
+        var timer = new FakeClock();
+
+        // Act
+        Func<Task> action = () => subject.Should(timer).CompleteWithinAsync(1.Seconds());
+        subject.SetResult();
+        timer.Complete();
+
+        // Assert
+        await action.Should().NotThrowAsync();
+    }
+
+    [Fact]
+    public async Task When_TCS_did_not_complete_in_time_it_should_fail()
```

Comment:
🤔 You're not completing the TCS at all. Was that intended?

Reply:
Yes. Why not?